File: Logic_HandsOn/C++/Problems on Digits,String,Array,Pointers etc/program12.cpp
#include<iostream>
using namespace std;

class Display
{
	private:
		int iValue;

	public:

		Display(int iValue)
		{
			this->iValue = iValue;
		}	

		bool CheckZero()
		{
			bool bAns=false;
			while(iValue!=0)
			{
				int iDigit = iValue%10;
				if(iDigit==0)
				{
					bAns=true;
					break;
				}
				iValue = iValue/10;
			}

			return bAns;
		}
};

int main()
{
	int iValue=0;
	bool bRet=false;

	cout<<"Enter Number: ";
	cin>>iValue;

	Display *dObj = new Display(iValue);
	bRet = dObj->CheckZero();

	if(bRet==true)
	{
		cout<<"Zero is present\n";
	}
	else
	{
		cout<<"Zero is Absent\n";
	}
	return 0;
}

/*
OUTPUT:

Enter Number: 2395
Zero is Absent

Enter Number: 1018
Zero is present

*/

File: Logic_HandsOn/Java/Problems on Array,Pattern etc/Program12.java
import java.util.*;

class Solution
{
	public int iValue;

	public Solution(int iValue)
	{
		this.iValue = iValue;
	}
                
	boolean CheckZero() 
  	{
		boolean bAns=false;
		while(iValue!=0)
		{
			int iDigit = iValue%10;
			if(iDigit==0)
			{
				bAns=true;
				break;
			}
			iValue = iValue/10;
		}
		return bAns;
	}
}

class Program2
{
	public static void main(String[] args)
	{
		Scanner sObj = new Scanner(System.in);

		boolean bAns=false;
		System.out.println("Enter number: ");
		int iNo = sObj.nextInt();

		Solution obj = new Solution(iNo);
		bAns = obj.CheckZero();

		if(bAns=false)
		{
			System.out.println("Zero is Absent");
		}
		else
		{
			System.out.println("Zero is Present");
		}	
	}
}

/*
OUTPUT:

Enter number: 
1018
Zero is Present

*/
